extensions/amp-consent/0.1/consent-ui.js

```diff
@@ -174,6 +182,47 @@ export class ConsentUI {
     if (this.isCreatedIframe_) {
       this.resetIframe_();
     }
+
+    this.isVisible_ = false;
+  }
+
+  /**
+   * Enter the fullscreen state for the UI
+   */
+  enterFullscreen() {
+    if (!this.ui_ || !this.isVisible_ || this.isFullscreen_) {
+      return;
+    }
+
+    const {classList} = this.parent_;
+    classList.add('i-amphtml-consent-ui-fullscreen');
+
+    this.isFullscreen_ = true;
+  }
+
+  /**
+   * Enter the fullscreen state for the UI
+   */
+  exitFullscreen() {
+    if (!this.ui_ || !this.isVisible_ || !this.isFullscreen_) {
+      return;
+    }
+
+    const {classList} = this.parent_;
+    classList.add('i-amphtml-consent-ui-fullscreen-exit');
+    classList.remove('i-amphtml-consent-ui-fullscreen');
+
+    this.isFullscreen_ = false;
+
+    // Set a timeout to remove the fullscreen exit class
+    // This timeout should match our CSS animation speed.
```

Comment:
Fixed the comment in my current local branch.

So the class removal is required to stop any jumping of the iframe inside the `amp-consent`, it looks really bad. If you look at the CSS, I couldn't think of another way to transition the height, with graphics accelerated CSS, and look nice without the `-exit` class.

I decided on the `transitionend`, because using a setTimeout just seemed incorrect and hack-y.  I didn't consider a user adding a transition themselves. But I have an idea, the [`transitionend event`](https://developer.mozilla.org/en-US/docs/Web/Events/transitionend) expsoses a property, perhaps we can use that in order to ensure that it is our transition, and then act accordingly?

Also, since this is our agreed UX usage of the component, perhaps it would be good to use some `!important` for the animations? Let me know what you think 😄

Reply:
Also, talked with another UXE, showed them my approach to the problem, and agreed this is probably the best/good way of doing this. We couldn't come up with another CSS solution to achieve this same effect.